scripts/ball.gd

```diff
@@ -16,13 +18,30 @@ func _fixed_process(delta):
 	for body in colliding_bodies:
 		if body.is_in_group("Bricks"):
 			get_node("/root/World").score += points_on_hit
-			body.queue_free()
+			body.collapse_and_maybe_drop_power_up()
+
 		elif body.get_name() == "Paddle":
 			var speed     = get_linear_velocity().length()
 			var direction = get_pos() - body.get_node("Kickback").get_global_pos()
 			var velocity  = direction.normalized() * min(speed + speedup, max_speed)
 			set_linear_velocity(velocity)
-			
+
+	if get_node("/root/World").power_up_time_left_dict.has("TeleControl"):
+		var velocity_vec = get_linear_velocity()
+		var velocity_x = velocity_vec.x
+		var velocity_y = velocity_vec.y
+		
+		# Calculating the acceleration multiplier
+		var ball_x_pos = get_pos().x
+		var paddle_x_pos = get_node("/root/World/Paddle").get_pos().x
+		var screen_x_size = get_viewport().get_rect().size.x
+		var acc_multiplier = (paddle_x_pos - ball_x_pos) / screen_x_size
```

Comment:
What do you think of changing this name to `acceleration_multiplier`, or `acceleration_rate`? Seems more descriptive.

Maybe change `ball_x_pos` to `ball_pos_x`, to repeat the `get_pos().x` and `velocity_vec.x` structures. :smiley: